internal/message/types_test.go

```diff
@@ -0,0 +1,122 @@
+package message
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	gwapiv1b1 "sigs.k8s.io/gateway-api/apis/v1beta1"
+
+	"github.com/envoyproxy/gateway/internal/ir"
+)
+
+func TestProviderResources(t *testing.T) {
```

Comment:
Can you add a test case where resources is nil?

xref: https://github.com/envoyproxy/gateway/pull/216